supervision/detection/line_counter.py

```diff
@@ -54,40 +99,87 @@ def trigger(self, detections: Detections) -> Tuple[np.ndarray, np.ndarray]:
         crossed_in = np.full(len(detections), False)
         crossed_out = np.full(len(detections), False)
 
-        for i, (xyxy, _, confidence, class_id, tracker_id) in enumerate(detections):
-            if tracker_id is None:
-                continue
+        if self.count_condition == "whole_crossed":
+            for i, (xyxy, _, confidence, class_id, tracker_id) in enumerate(detections):
+                if tracker_id is None:
+                    continue
 
-            x1, y1, x2, y2 = xyxy
-            anchors = [
-                Point(x=x1, y=y1),
-                Point(x=x1, y=y2),
-                Point(x=x2, y=y1),
-                Point(x=x2, y=y2),
-            ]
-            triggers = [self.vector.is_in(point=anchor) for anchor in anchors]
+                x1, y1, x2, y2 = xyxy
 
-            if len(set(triggers)) == 2:
-                continue
+                anchors = [
+                    Point(x=x1, y=y1),
+                    Point(x=x1, y=y2),
+                    Point(x=x2, y=y1),
+                    Point(x=x2, y=y2),
+                ]
 
-            tracker_state = triggers[0]
+                triggers = [
+                    (self.vector.cross_product(point=anchor) < 0) for anchor in anchors
+                ]
+
+                if len(set(triggers)) == 2:
+                    continue
+
+                tracker_state = triggers[0]
+
+                if tracker_id not in self.tracker_state:
+                    self.tracker_state[tracker_id] = tracker_state
+                    continue
+
+                if self.tracker_state.get(tracker_id) == tracker_state:
+                    continue
 
-            if tracker_id not in self.tracker_state:
                 self.tracker_state[tracker_id] = tracker_state
-                continue
 
-            if self.tracker_state.get(tracker_id) == tracker_state:
-                continue
+                all_anchors_in_range = True
+                for anchor in anchors:
+                    if not self.is_point_in_line_range(anchor):
+                        all_anchors_in_range = False
+                        break
+
+                if not all_anchors_in_range:
+                    continue
+
+                if tracker_state:
+                    self.in_count += 1
+                    crossed_in[i] = True
+                else:
+                    self.out_count += 1
+                    crossed_out[i] = True
+
+            return self.in_count, self.out_count
+
+        elif self.count_condition == "center_point_crossed":
+            for i, (xyxy, _, confidence, class_id, tracker_id) in enumerate(detections):
+                if tracker_id is None:
+                    continue
+
+                x1, y1, x2, y2 = xyxy
+
+                # Calculate the center point of the box
+                center_point = Point(x=(x1 + x2) / 2, y=(y1 + y2) / 2)
```

Comment:
Let's not limit ourselves to just the center of the box. Let's make it more general. Also, you can use [`get_anchors_coordinates`](https://github.com/roboflow/supervision/blob/64661a3e4aae2830a37cc277ec2d73c9dee0ee1c/supervision/detection/core.py#L760) to calculate coordinates of specific bounding box points. (already tested with multiple unit tests)

Reply:
I have used the function you mentioned in the new commit.